lint/verilator_waiver.vlt

```diff
@@ -58,6 +58,40 @@ lint_off -rule UNUSED -file "*/rtl/sim/simulator_ctrl.sv" -match "*'wdata_i'[31:
 // entire 32-bit address around to make the code a bit cleaner.
 lint_off -rule UNUSED -file "*/rtl/timer.sv" -match "*'timer_addr_i'[31:10]*"
 
+// Bits of signal are not used for MHPMCounterNum < 29: mhpmcounter_we[31:MHPMCounterNum+3]
+// cleaner to write all bits even if not all are used
+lint_off -rule UNUSED -file "*/rtl/ibex_cs_registers.sv" -lines 190
+
+// Bits of signal are not used for MHPMCounterNum < 29: mhpmcounterh_we[31:MHPMCounterNum+3]
+// cleaner to write all bits even if not all are used
+lint_off -rule UNUSED -file "*/rtl/ibex_cs_registers.sv" -lines 191
+
+// Bits of signal are not used for MHPMCounterNum < 29: mhpmcounter_incr[31:MHPMCounterNum+3]
+// cleaner to write all bits even if not all are used
+lint_off -rule UNUSED -file "*/rtl/ibex_cs_registers.sv" -lines 192
+
+// Signals are unused if MHPMCounterNum == 0: clk_i, rst_ni
+// If no counters are implemented, no clock and no reset is needed.
+lint_off -rule UNUSED -file "*/rtl/ibex_counters.sv" -lines 6
+lint_off -rule UNUSED -file "*/rtl/ibex_counters.sv" -lines 7
+
+// Signal is unused if MHPMCounterNum == 0: counter_val_i[31:0]
+// If no counters are implemented, no next-state logic is elaborated.
+lint_off -rule UNUSED -file "*/rtl/ibex_counters.sv" -lines 12
+
+// Bits of signal are not used for MHPMCounterNum < 29: counter_inc_i[29:MHPMCounterNum]
+// cleaner to write all bits even if not all are used
+lint_off -rule UNUSED -file "*/rtl/ibex_counters.sv" -lines 9
+
+// Bits of signal are not used for MHPMCounterNum < 29: counterh_we_i[29:MHPMCounterNum]
+// cleaner to write all bits even if not all are used
+lint_off -rule UNUSED -file "*/rtl/ibex_counters.sv" -lines 10
+
+// Bits of signal are not used for MHPMCounterNum < 29: counter_we_i[29:MHPMCounterNum]
+// cleaner to write all bits even if not all are used
+lint_off -rule UNUSED -file "*/rtl/ibex_counters.sv" -lines 11
```

Comment:
Can you please change to the `-match` syntax here? As opposed to specifying the line number, this requires less work to keep it up to date with the RTL and is supported as of Verilator v.4.028.

Reply:
done .
Is there a way to parameterize the waiver (ignore bits `[63:MHPMCounterWidth]` for example)?
I did not find a way, except for wildcards, which in the end is actually the same thing as ignoring the whole signal. - maybe I should just do that?

Reply:
Thanks for changing the waiver rules.
> Is there a way to parameterize the waiver (ignore bits [63:MHPMCounterWidth] for example)? I did not find a way, except for wildcards, which in the end is actually the same thing as ignoring the whole signal. - maybe I should just do that?

I don't know if there is a way evaluate the parameter to generate the matching string. I think what you came up with is alright.

Reply:
OK, thank you.

I had to change it for `counter_inc_i`, `counter_we_i` and `counterh_we_i`, since for `MHPMCounterNum=0` the whole signal must be ignored and apparently the corresponding warning is not captured by ignoring all bits.

Reply:
SGTM, I am resolving this one.